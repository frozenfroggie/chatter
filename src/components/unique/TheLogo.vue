<template>
  <div class='logoContainer'>
    <div class='logo'>
      <div class='logoInner'>
      </div>
    </div>
    <div class='logoText'>
      <h1> CHATTER </h1>
    </div>
  </div>
</template>

<script>
export default {
}
</script>

<style scoped>
.logoContainer {
  grid-area: logoContainer;
  display: flex;
  transform: scale(0.5);
}
.logoText {
  color: rgba(250,250,250,0.9);
  padding-left: 20px;
}
.logo {
  width: 70px;
  height: 70px;
  border: 6px solid rgba(250,250,250,0.9);
  border-radius: 50%;
}
.logo:after {
  display: block;
  content: '';
  position: absolute;
  left: 51px;
  top: 51px;
  width: 20px;
  height: 20px;
  background-color: rgb(46, 59, 79);
  transform: skew(20deg,20deg);
}
.logo:before {
  display: block;
  content: '';
  position: absolute;
  left: 59px;
  top: 59px;
  width: 20px;
  height: 20px;
  background-color: rgba(250,250,250,0.9);
  transform: skew(20deg,20deg);
}
.logoInner {
  position: absolute;
  width: 34px;
  height: 34px;
  border: 5px solid rgba(250,250,250,0.9);
  border-radius: 50%;
  left: 20px;
  top: 18px;
  z-index: 1;
}
.logoInner:after {
  display: block;
  content: '';
  position: absolute;
  left: 1px;
  top: 18px;
  width: 14px;
  height: 14px;
  background-color: rgb(46, 59, 79);
  transform: skew(-20deg,-20deg);
}
.logoInner:before {
  display: block;
  content: '';
  position: absolute;
  left: -5px;
  top: 25px;
  width: 13px;
  height: 13px;
  background-color: rgba(250,250,250,0.9);
  transform: skew(-20deg,-20deg);
}
</style>
