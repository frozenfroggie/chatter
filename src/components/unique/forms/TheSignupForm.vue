<template>
  <div class="inputContainer">
    <div :class="['username', {invalid: $v.username.$error && username !== ''}]">
      <icon name="user" scale="1"></icon>
      <input
        type="text"
        name="username"
        placeholder="Username"
        aria-label="Username"
        id="username"
        :value="username"
        @input="(event) => updateSignupForm(event, 'username')"
        @blur="$v.username.$touch()">
    </div>
    <div :class="['email', {invalid: $v.email.$error && email !== ''}]">
      <icon name="envelope" scale="1"></icon>
      <input
        type="email"
        name="email"
        placeholder="Email"
        aria-label="Email"
        id="email"
        :value="email"
        @input="(event) => updateSignupForm(event, 'email')"
        @blur="$v.email.$touch()">
    </div>
    <div :class="['password', {invalid: $v.password.$error && password !== ''}]">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Password"
        aria-label="Password"
        id="password"
        :value="password"
        @input="(event) => updateSignupForm(event, 'password')"
        @blur='$v.password.$touch()'>
    </div>
    <div :class="['confirmPassword', {invalid: $v.confirmPassword.$error && confirmPassword !== ''}]">
      <icon name="unlock" scale="1"></icon>
      <input
        type="password"
        name="password"
        placeholder="Retype Password"
        aria-label="Retype password"
        id="confirmPassword"
        :value="confirmPassword"
        @input="(event) => updateSignupForm(event, 'confirmPassword')"
        @blur='$v.confirmPassword.$touch()'>
    </div>
  </div>
</template>

<script>
import { required, email, minLength, sameAs } from 'vuelidate/lib/validators'

export default {
  computed: {
    username () {
      return this.$store.getters.signupForm.username
    },
    email () {
      return this.$store.getters.signupForm.email
    },
    password () {
      return this.$store.getters.signupForm.password
    },
    confirmPassword () {
      return this.$store.getters.signupForm.confirmPassword
    }
  },
  methods: {
    updateSignupForm (event, name) {
      this.$store.dispatch('updateSignupForm', {name, value: event.target.value})
    }
  },
  validations: {
    username: {
      required,
      minLen: minLength(5)
    },
    email: {
      required,
      email
    },
    password: {
      required,
      minLen: minLength(6)
    },
    confirmPassword: {
      sameAs: sameAs('password')
    }
  }
}
</script>

<style scoped>
.inputContainer {
  grid-area: loginInput;
  display: grid;
  grid-template-rows: 1fr 1fr 1fr 1fr;
  grid-template-areas:
    ' username '
    ' email '
    ' password '
    ' confirmPassword ';
  background-color: rgb(38, 49, 66);
}
.email {
  grid-area: email;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.username {
  grid-area: username;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.password {
  grid-area: password;
  border-top: 1px solid rgba(250,250,250,0.5);
}
.confirmPassword {
  grid-area: confirmPassword;
  border-top: 1px solid rgba(250,250,250,0.5);
  border-bottom: 1px solid rgba(250,250,250,0.5);
}
input {
  height: 100%;
  width: 80%;
  padding-left: 10px;
  background-color: inherit;
  border: none;
  color: rgba(250,250,250,0.9);
}
.invalid input {
  color: #cb2431;
}
@media only screen and (min-width: 768px) {
  .inputContainer {
    border-radius: 10px;
  }
  .username {
    border-radius: 10px 10px 0px 0px;
  }
  .confirmPassword {
    border-radius: 0px 0px 10px 10px;
  }
}
</style>
