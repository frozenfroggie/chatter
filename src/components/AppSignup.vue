<template>
  <div class="login">

    <the-logo />

    <div class='loginHeader'> Create your account </div>

    <the-signup-form />

    <button type="button" name="" class='loginBtn' @click="signup"> SIGN UP </button>

    <div class='signin' @click='$router.push("/")'> Already have an account? <span> Sign in </span> </div>

  </div>
</template>

<script>
import { mapActions } from 'vuex'
import TheLogo from './unique/TheLogo.vue'
import TheSignupForm from './unique/forms/TheSignupForm.vue'

export default {
  components: {
    TheLogo,
    TheSignupForm
  },
  methods: {
    ...mapActions(['signup'])
  }
}
</script>

<style scoped>
.login {
  position: absolute;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  color: rgba(250,250,250,0.9);
  display: grid;
  grid-template-columns: 2fr 1fr;
  grid-template-rows: 1fr 100px 40px 200px 80px 2fr 50px;
  grid-template-areas:
    ' . . '
    ' logoContainer logoContainer'
    ' loginHeader loginHeader '
    ' loginInput loginInput'
    ' loginBtn .'
    ' . . '
    ' signin signin ';
}
.loginHeader {
  grid-area: loginHeader;
  text-align: left;
  padding-left: 10%;
}
.loginBtn {
  grid-area: loginBtn;
  margin: 1.2em 1em;
  width: 80%;
  color: rgba(250,250,250,0.9);
  background-color: rgb(38, 49, 66);
  border: 1px solid rgba(250,250,250,0.5);
  border-radius: 22px;
}
.loginBtn:hover {
  cursor: pointer;
}
.signin {
  grid-area: signin;
  font-size: .85em;
}
.signin > span {
  font-weight: 600;
}
.signin > span:hover {
  cursor: pointer;
}
@media only screen and (min-width: 768px) {
  .login {
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 120px 50px 200px 80px 2fr 50px;
    grid-template-areas:
      ' . . . . '
      ' . logoContainer logoContainer .'
      ' . loginHeader loginHeader . '
      ' . loginInput loginInput . '
      ' . loginBtn forgotPassword . '
      ' . . . . '
      ' . signin signin .';
  }
}
@media only screen and (min-width: 1024px) {
  .login {
    grid-template-columns: 30% 1fr 1fr 30%;
  }
}
@media only screen and (min-width: 1280px) {
  .login {
    grid-template-columns: 32% 1fr 1fr 32%;
  }
}
</style>
