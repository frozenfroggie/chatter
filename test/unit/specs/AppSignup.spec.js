import { shallowMount, mount } from '@vue/test-utils'
import AppSignup from '@/components/AppSignup'
import TheLogo from '@/components/unique/TheLogo'
import TheSignupForm from '@/components/unique/forms/TheSignupForm'

describe('AppSignup', () => {
  it('renders TheLogo component', () => {
    const wrapper = shallowMount(AppSignup)
    const TheLogoComponent = wrapper.find(TheLogo)
    expect(TheLogoComponent.is(TheLogo)).toBe(true)
  })
  it('renders TheSignupForm component', () => {
    const wrapper = shallowMount(AppSignup)
    const TheSignupFormComponent = wrapper.find(TheSignupForm)
    expect(TheSignupFormComponent.is(TheSignupForm)).toBe(true)
  })
  it('matches snapshot', () => {
    const wrapper = shallowMount(AppSignup)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
