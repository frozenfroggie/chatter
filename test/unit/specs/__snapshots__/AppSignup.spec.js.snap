// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSignup matches snapshot 1`] = `
<div class="login">
    <!---->
    <div class="loginHeader"> Create your account </div>
    <!---->
    <button type="button" name="" class="loginBtn"> SIGN UP </button>
    <div class="signin"> Already have an account? <span> Sign in </span></div>
</div>
`;
